test/e2e/specs/delete-prohibited/recordset/facet-presentation.spec.ts

```diff
@@ -0,0 +1,249 @@
+  sort: '@sort(id)',
+  totalNumFacets: 23,
+  facetNames: [
+    'id', 'int_col', 'float_col', 'date_col', 'timestamp_col', 'text_col',
+    'longtext_col', 'markdown_col', 'boolean_col', 'jsonb_col', 'F1',
+    'to_name', 'f3 (term)', 'from_name', 'F1 with Term', 'Check Presence Text',
+    'F3 Entity', 'F5', 'F5 with filter', 'Outbound1 (using F1)',
+    'col_w_column_order_false', 'col_w_column_order', 'col_w_long_values'
+  ],
+  defaults: {
+    openFacetNames: ['id', 'int_col', 'to_name'],
+    numFilters: 2,
+    numRows: 1,
+    pageSize: 25
+  },
+  searchBox: {
+    term: 'one',
+    numRows: 6,
+    term2: 'eve',
+    term2Rows: 4,
+    term3: 'evens',
+    term3Rows: 1
+  },
+  facetComments: [
+    'ID comment', 'int comment', null, null, 'timestamp column', null,
+    'A lengthy comment for the facet of the longtext_col. This should be displyed properly in the facet.',
+    null, null, null, null, null, null, null, 'F1 with Term comment', null, null,
+    null, 'has filters', 'is using another facet sourcekey in source', null
+  ]
+};
+
+test('Viewing Recordset with Faceting, default presentation based on facets annotation', async ({ page, baseURL }, testInfo) => {
+  await test.step('should load recordset page', async () => {
+    const PAGE_URL = `/recordset/#${getCatalogID(testInfo.project.name)}/${testParams.schema_name}:${testParams.table_name}${testParams.sort}`;
+
+    await page.goto(`${baseURL}${PAGE_URL}`);
+    await RecordsetLocators.waitForRecordsetPageReady(page);
+  });
+
+  await test.step('should have 1 row visible', async () => {
+    await expect.soft(RecordsetLocators.getRows(page)).toHaveCount(testParams.defaults.numRows);
+  });
+
+  await test.step('verify the text is truncated properly on load, then not truncated after clicking "more"', async () => {
+    // default config: maxRecordsetRowHeight = 100
+    // 100 for max height, 10 for padding, 1 for border
+    const cellHeight = 110;
+    let testCellDimensions;
+
+    const firstRow = RecordsetLocators.getRows(page).nth(0);
+    const testCell = RecordsetLocators.getRowCells(firstRow).nth(2);
+
+    await expect.soft(testCell).toContainText('... more');
+    testCellDimensions = await testCell.boundingBox();
+    expect.soft(Math.trunc(testCellDimensions?.height || 0)).toBe(cellHeight);
+
+    await testCell.locator('.readmore').click();
+
+    await expect.soft(testCell).toContainText('... less');
+    testCellDimensions = await testCell.boundingBox();
+    expect.soft(testCellDimensions?.height).toBeGreaterThan(cellHeight);
+  });
+
+  await test.step('should have ' + testParams.totalNumFacets + ' facets', async () => {
+    await expect.soft(RecordsetLocators.getAllFacets(page)).toHaveCount(testParams.totalNumFacets);
+
+    const titles = RecordsetLocators.getFacetTitles(page);
+    const count = await titles.count();
+    for (let i = 0; i < count; i++) {
+      await expect.soft(titles.nth(i)).toHaveText(testParams.facetNames[i])
```

Comment:
toHaveText handles array of elements. So just the following works too:
```tsx
await expect.soft(titles).toHaveText(testParams.facetNames);
```

In other places that I had to do something similar (like [here](https://github.com/informatics-isi-edu/chaise/blob/master/test/e2e/specs/all-features/record/related-table.spec.ts#L75,L76)) I usually test the count separately just so it's easier to debug.

Reply:
While I think what you have is fine, we should try to avoid using locator functions like `count()` if we can. That's what the official document says as well

https://playwright.dev/docs/api/class-locator#locator-count
>If you need to assert the number of elements on the page, prefer [expect(locator).toHaveCount()](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count) to avoid flakiness. See [assertions guide](https://playwright.dev/docs/test-assertions) for more details.


P.S. I'm not saying that we shouldn't use `.count()` at all. I cases like the `testRangeInputSubmitThenClear` function I think using `count()` makes sense, as it will allow us to avoid hardcoding the number. But in cases where we already know the count, or if we just want to eventually use `toHaveText` we should try to avoid these flaky functions.